modules/openthread/platform/crypto_psa.c

```diff
@@ -418,3 +424,137 @@ otError otPlatCryptoRandomGet(uint8_t *aBuffer, uint16_t aSize)
 {
 	return psaToOtError(psa_generate_random(aBuffer, aSize));
 }
+
+#if defined(CONFIG_OPENTHREAD_ECDSA)
+
+otError otPlatCryptoEcdsaGenerateKey(otPlatCryptoEcdsaKeyPair *aKeyPair)
+{
+	psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+	psa_key_id_t key_id;
+	psa_status_t status;
+	size_t exported_length;
+
+	psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_EXPORT);
+	psa_set_key_algorithm(&attributes, PSA_ALG_DETERMINISTIC_ECDSA(PSA_ALG_SHA_256));
+	psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&attributes, 256);
+
+	status = psa_generate_key(&attributes, &key_id);
+	if (status != PSA_SUCCESS) {
+		goto out;
+	}
+
+	status = psa_export_key(key_id, aKeyPair->mDerBytes, OT_CRYPTO_ECDSA_MAX_DER_SIZE,
```

Comment:
It's a pity there's no API to destroy the key. Then we could just store the key id inside aKeyPair and use it more securely. I guess we can propose such addition in the future.